### PR TITLE
ci: introduce the internal DD_TESTING_RAISE variable

### PR DESCRIPTION
**Pull request by P403n1x87**
## Description

This variable is meant to be used to control exceptional behaviour and whether to let the exception bubble up or handle it and turn it into a warning.

## Checklist
- [ ] Added to the correct milestone.
- [ ] Tests provided or description of manual testing performed is included in the code or PR.
- [ ] Library documentation is updated.
- [ ] [Corp site](https://github.com/DataDog/documentation/) documentation is updated (link to the PR).

### CODE REVIEW

**Review by brettlangdon — COMMENTED** [review]
I am not sure we want to set `DD_ENV` for all tests, it might make it harder to validate specific behaviors with/without the env variable set? (maybe that indicates bad tests?)

We should be able to use CircleCI env variables to determine if we are in a test.

https://circleci.com/docs/2.0/env-vars/#built-in-environment-variables

e.g. `CI=true` which should also be easy enough to set locally? (although will that have any impact on `pytest` contrib tests?)

Or else, we can just go full custom env variable, `DD_DEV=true` or... something unused/not likely to be used by us.

**Comment by P403n1x87** [comment]
> Or else, we can just go full custom env variable, `DD_DEV=true` or... something unused/not likely to be used by us.
I had similar concerns and was about to introduce `DD_TRACE_ENV` but looking around it seemed that tests would override the value of `DD_ENV` when needed so I thought there was no need for a new variable. `CI` sounds like a good alternative 👍 . I'll use that instead.

**Comment by P403n1x87** [comment]
From offline discussion we agreed that a dedicated variable is also fine.

File: .circleci/config.yml [hunk 1]
```diff
@@ -127,6 +127,7 @@ commands:
             - run:
                 environment:
                   DD_TRACE_AGENT_URL: http://localhost:9126
+                  DD_TESTING_RAISE: true
```
Comment (brettlangdon) [inline .circleci/config.yml:130]: do we need `--pass-env` on `riot` for the `unless` condition ?

Comment (P403n1x87) [reply .circleci/config.yml:130]: ah nice catch! I think we'd need it; is there a reason why it's not currently used there?

Comment (brettlangdon) [reply .circleci/config.yml:130]: not that I can think of

Comment (brettlangdon) [reply .circleci/config.yml:130]: https://github.com/DataDog/dd-trace-py/pull/2500

**Review by Kyle-Verhoog — APPROVED** [review]

**Review by brettlangdon — APPROVED** [review]

**Pull request merged**